core/src/main/scala/kafka/utils/IteratorTemplate.scala

```diff
@@ -27,7 +29,7 @@ object FAILED extends State
  * Transliteration of the iterator template in google collections. To implement an iterator
  * override makeNext and call allDone() when there is no more items
  */
-abstract class IteratorTemplate[T] extends Iterator[T] with java.util.Iterator[T] {
+abstract class IteratorTemplate[T] extends Iterator[T] with ClosableIterator[T] {
```

Comment:
Do we need this? It seems to me that some `IteratorTemplate` classes won't necessarily have a `close` method.